src/lightning/data/streaming/dataset.py

```diff
@@ -53,12 +74,19 @@ def __init__(
         if not isinstance(shuffle, bool):
             raise ValueError(f"Shuffle should be a boolean. Found {shuffle}")
 
-        self.cache = Cache(name=name, version=version, cache_dir=cache_dir, item_loader=item_loader, chunk_bytes=1)
+        input_dir = _resolve_dir(input_dir)
+
+        # Override the provided input_path
+        cache_dir = _try_create_cache_dir()
+        if cache_dir:
+            input_dir.path = cache_dir
+
+        self.cache = Cache(input_dir=input_dir, item_loader=item_loader, chunk_bytes=1)
 
         self.cache._reader._try_load_config()
 
         if not self.cache.filled:
-            raise ValueError(f"The provided dataset `{name}` isn't filled up.")
+            raise ValueError(f"The provided dataset `{input_dir}` isn't filled up.")
```

Comment:
This is a pretty cryptic error message IMO. It's probably not likely to be happen for actual users but a bit more context could help.

Reply:
Done !